toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/popup/PopupViewTestActivity.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.test.popup
+
+import android.os.Bundle
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import com.esri.arcgisruntime.data.Feature
+import com.esri.arcgisruntime.geometry.GeometryType
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.toolkit.extension.logTag
+import com.esri.arcgisruntime.toolkit.popup.PopupView
+import com.esri.arcgisruntime.toolkit.popup.PopupViewModel
+import com.esri.arcgisruntime.toolkit.test.R
+import com.esri.arcgisruntime.toolkit.test.databinding.ActivityPopupBinding
+import com.esri.arcgisruntime.toolkit.util.observeEvent
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import kotlin.math.roundToInt
+
+/**
+ * Activity to show usages of [PopupView] and [PopupViewModel].
```

Comment:
This class level doc could have some more detail on explaining the implementation:
- the activity's layout hosts the MapView and bottom sheet
- creates and owns the PopupViewModel for showing the Popup of an identified feature
- it handles tap gestures on the MapView in order to identify a feature by overriding onSingleTapConfirmed on DefaultMapViewOnTouchListener
- when a feature is identified, selects that feature in the MapView, initializes the PopupViewModel and shows the bottom sheet populated with that identified feature 
- Could also explain here what happens when the user tabs on the feature in the bottom sheet, i.e. showing the PopupView...

Reply:
Updated.